detection_rules/rule_validators.py

```diff
@@ -93,68 +93,111 @@ def validate_stack_combos(self, data: QueryRuleData, meta: RuleMeta) -> Union[KQ
                 print(err_trailer)
                 return exc
 
-    def validate_integration(self, data: QueryRuleData, meta: RuleMeta, package_integrations: List[dict]) -> Union[
-            KQL_ERROR_TYPES, None, TypeError]:
+    def validate_integration(
+        self, data: QueryRuleData, meta: RuleMeta, package_integrations: List[dict]
+    ) -> Union[KQL_ERROR_TYPES, None, TypeError]:
         """Validate the query, called from the parent which contains [metadata] information."""
         if meta.query_schema_validation is False or meta.maturity == "deprecated":
-            # syntax only, which is done via self.ast
             return
 
         error_fields = {}
-        current_stack_version = ""
-        combined_schema = {}
-        for integration_schema_data in get_integration_schema_data(data, meta, package_integrations):
-            ecs_version = integration_schema_data['ecs_version']
-            integration = integration_schema_data['integration']
-            package = integration_schema_data['package']
-            package_version = integration_schema_data['package_version']
-            integration_schema = integration_schema_data['schema']
-            stack_version = integration_schema_data['stack_version']
-
-            if stack_version != current_stack_version:
-                # reset the combined schema for each stack version
-                current_stack_version = stack_version
-                combined_schema = {}
-
-            # add non-ecs-schema fields for edge cases not added to the integration
+        package_schemas = {}
+
+        # Initialize package_schemas with a nested structure
+        for integration_data in package_integrations:
+            package = integration_data["package"]
+            integration = integration_data["integration"]
```

Comment:
```suggestion
            package = integration_data.get("package", None)
            integration = integration_data.get("integration", None)
```

Nit. The `setdefault` on the following lines will handle if these values return None; however, might there be a case there "package" or "integration" may not be set? This is probably enforced before here via the schema, but may want to use `.get` in case.